### PR TITLE
Feature/dalantren/add explicit sort mode

### PR DESCRIPTION
**Pull request by Dalantren**

### CODE REVIEW

File: src/examples/sort-row-explicit/example-sort-row-explicit.component.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<q-grid [caption]="title"
+        [rows]="rows | async"
+        [sortMode]="'explicit'"
```
Comment (klumba12) [inline src/examples/sort-row-explicit/example-sort-row-explicit.component.html:3]: you can just write sortMode="explicit"

File: projects/qgrid-core/sort/sort.state.d.ts [hunk 2]
```diff
@@ -3,8 +3,9 @@
  *
  * * `'multiple'` allows to sort by several column keys.
  * * `'single'` allows to sort only by one column key.
+ * * `'explicit'` allows to sort multiple columns while Shift key pressed otherwise applies single mode
  */
-export declare type SortStateMode = 'single' | 'multiple';
+export declare type SortStateMode = 'single' | 'multiple' | 'explicit';
```
Comment (klumba12) [inline projects/qgrid-core/sort/sort.state.d.ts:8]: what other variants do you have for the name?

File: projects/qgrid-core/sort/sort.let.js [hunk 3]
```diff
@@ -20,7 +20,15 @@ export class SortLet {
 				const key = column.key;
 				const sort = model.sort;
 				const sortState = sort();
-				const by = Array.from(sortState.by);
+				let by = Array.from(sortState.by);
+
+				if (sortState.mode === 'explicit') {
+					const { code, status } = model.keyboard();
+					if (code !== 'shift' || status !== 'down') {
```
Comment (klumba12) [inline projects/qgrid-core/sort/sort.let.js:27]: can you make a named variable for this boolean?

File: projects/qgrid-core/sort/sort.state.d.ts (hunk 2 above)
Comment (klumba12) [reply projects/qgrid-core/sort/sort.state.d.ts:8]: also please make this mode by default and make the `trigger` property empty.

Comment (Dalantren) [reply projects/qgrid-core/sort/sort.state.d.ts:8]: what about 'multipleKey' or 'keyMultiple'?
Also it could be 'mixed'

Comment (klumba12) [reply projects/qgrid-core/sort/sort.state.d.ts:8]: yes great let's rename it to mixed.

File: projects/qgrid-core/sort/sort.let.js [hunk 4]
```diff
@@ -20,7 +20,17 @@ export class SortLet {
 				const key = column.key;
 				const sort = model.sort;
 				const sortState = sort();
-				const by = Array.from(sortState.by);
+				let by = Array.from(sortState.by);
+
+				if (sortState.mode === 'explicit') {
+					const { code, status } = model.keyboard();
+					const isKeyPressed = code === 'shift' && status === 'down';
```
Comment (klumba12) [inline projects/qgrid-core/sort/sort.let.js:27]: the name is bad, it doesn't say what is going on

**Review by klumba12 — APPROVED** [review]

**Pull request merged**